dht.go

```diff
@@ -507,3 +578,46 @@ func (dht *IpfsDHT) newContextWithLocalTags(ctx context.Context, extraTags ...ta
 	) // ignoring error as it is unrelated to the actual function of this code.
 	return ctx
 }
+
+func (dht *IpfsDHT) handleProtocolChanges(ctx context.Context) {
+	// register for event bus protocol ID changes
+	sub, err := dht.host.EventBus().Subscribe(new(event.EvtPeerProtocolsUpdated))
+	if err != nil {
+		panic(err)
+	}
+	defer sub.Close()
+
+	pmap := make(map[protocol.ID]bool)
+	for _, p := range dht.protocols {
+		pmap[p] = true
+	}
+
+	for {
+		select {
+		case ie, ok := <-sub.Out():
+			e, ok := ie.(event.EvtPeerProtocolsUpdated)
+			if !ok {
+				logger.Errorf("got wrong type from subscription: %T", ie)
+				return
+			}
+
+			protos, err := dht.peerstore.SupportsProtocols(e.Peer, dht.protocolStrs()...)
+			if err != nil {
+				continue
+			}
```

Comment:
it's not really clear what to do here since this shouldn't really happen unless the peerstore is remotely mounted or gets corrupted. Some options:
1) Skip the update (maybe throw in a log message)
2) Freak out and panic
3) Try to use the event info to decide what to do (e.g. use the code that was previously here utilizing e.Added and e.Removed)

Reply:
Log & skip it for now. But let's leave a comment. This _really_ shouldn't be a huge issue in practice.